### ISSUE TITLE
Knights asking for a bow

### ISSUE THREAD

**Issue by Rosomah** [ldtteam/minecolonies#1443]
### Minecolonies version

Version: 1.10.2-0.8.3896

### Expected behavior

- Should be happy with a sword.

### Actual behaviour

- Wants a bow instead.

**Comment by Asherslab** [comment]
@Raycoms apparently you made a mistake with the Barracks, i'm taking a look but i dunno if i'll find it before i go lol

**Comment by Raycoms** [comment]
Sure it is a knight, he has the knight skin, and there is no second guard
in the tower who might want it?

**Comment by Rosomah** [comment]
I'm sure it's a knight since I chose to manually select what class is in that tower and I selected knight. I don't know how to add more guards than one though. I have 2 towers with knight selected and they're both asking for a bow.

**Comment by Raycoms** [comment]
Uhm, after you manually selected this profession this won't switch them
automatically to knights. Only the next ones you will hire will become
knights.
So you might try to fire the current one and hire them again and they
should become knights and ask for swords.
We made that so you don't switch between the professions. A knight will
always stay a knight unless you fire him and hire him as a ranger.

2017-09-10 15:51 GMT+01:00 Rosomah <notifications@github.com>:

**Comment by Rosomah** [comment]
Yeah, I did that. I fired him, set the profession to knight and then rehired. He's still asking for a bow. Another thing that happened was, that everyone became unemployed when I loaded the world after updating the mod to this version. They had the clothes which belong to the job they were doing and they had the inventory the same, but I had to rehire everyone.
But when I rehired the knights, they lost all of their xp (this only happened to the knights).
